src/logaze/transform.clj

```diff
@@ -52,8 +52,8 @@
 
 (defn transform-attributes [attrs]
   (-> attrs
-      (extract :web-price :orig-price #(Float/parseFloat %))
-      (extract :final-price :price #(Float/parseFloat %))
+      (extract :web-price :orig-price #(Double/parseDouble %))
+      (extract :final-price :price #(Double/parseDouble %))
```

Comment:
When I wrote this, I chose a float since I don't expect prices to be huge numbers. Is there a reason why you chose double?

Reply:
When using `Float/parseFloat`, I was seeing this error with the tests:

```clojure
PS C:\src\logaze-scraper> lein eftest

FAIL in logaze.transform-test/regression-test (transform_test.clj:137)
expected: {:available true,
           :battery "Up to 14.1 hours",
           :brand "ThinkPad",
           :camera "720p HD",
           :display "13.3\" FHD (1920 x 1080) IPS, anti-glare, touchscreen, 300 nits",
           :final-price "372.29",
           :fingerprint-reader "Fingerprint Reader",
           :graphics "Integrated Intel┬« UHD Graphics",
           :inventory-status 1,
           :keyboard "Backlit - US English",
           :memory "8 GB DDR4 2667MHz (Soldered)",
           :memory-size "8GB",
           :memory-soldered true,
           :model "ThinkPad L13 Intel (13\") - Black",
           :operating-system "Windows 10 Pro 64",
           :orig-price 729.99,
           :percentage-savings nil,
           :price 372.29,
           :processor "10th Generation Intel┬« CoreΓäó i5-10210U Processor (1.60 GHz, up to 4.20 GHz with Turbo Boost, 4 Cores, 8 Threads, 6 MB Cache)",
           :processor-brand "Intel",
           :processor-cache "6M",
           :processor-range "i5",
           :product-code "20R3X001US",
           :product-condition "Refurbished",
           :product-mkt-name "ThinkPad L13 Intel (13\") - Black",
           :product-number "20R3X001US",
           :resolution "1920x1080",
           :screen-has-ips true,
           :screen-size 13.3,
           :storage "256 GB PCIe SSD",
           :storage-size "256GB",
           :storage-type "SSD",
           :touch-screen "Touch",
           :url "https://www.lenovo.com/us/outletus/en/p/laptops/thinkpad/thinkpadl/L13-Clam-2019/20R3X001US",
           :warranty "1 Year Depot or Carry-in",
           :web-price "729.99",
           :weight "Starting at 3.04 lbs (1.38 kg)",
           :wlan "Intel┬« 9560 802.11AC (2 x 2) & Bluetooth┬« 5.1"}
  actual: {:available true,
           :battery "Up to 14.1 hours",
           :brand "ThinkPad",
           :camera "720p HD",
           :display "13.3\" FHD (1920 x 1080) IPS, anti-glare, touchscreen, 300 nits",
           :final-price "372.29",
           :fingerprint-reader "Fingerprint Reader",
           :graphics "Integrated Intel┬« UHD Graphics",
           :inventory-status 1,
           :keyboard "Backlit - US English",
           :memory "8 GB DDR4 2667MHz (Soldered)",
           :memory-size "8GB",
           :memory-soldered true,
           :model "ThinkPad L13 Intel (13\") - Black",
           :operating-system "Windows 10 Pro 64",
           :orig-price 729.99,
           :percentage-savings nil,
           :price 372.29,
           :processor "10th Generation Intel┬« CoreΓäó i5-10210U Processor (1.60 GHz, up to 4.20 GHz with Turbo Boost, 4 Cores, 8 Threads, 6 MB Cache)",
           :processor-brand "Intel",
           :processor-cache "6M",
           :processor-range "i5",
           :product-code "20R3X001US",
           :product-condition "Refurbished",
           :product-mkt-name "ThinkPad L13 Intel (13\") - Black",
           :product-number "20R3X001US",
           :resolution "1920x1080",
           :screen-has-ips true,
           :screen-size 13.3,
           :storage "256 GB PCIe SSD",
           :storage-size "256GB",
           :storage-type "SSD",
           :touch-screen "Touch",
           :url "https://www.lenovo.com/us/outletus/en/p/laptops/thinkpad/thinkpadl/L13-Clam-2019/20R3X001US",
           :warranty "1 Year Depot or Carry-in",
           :web-price "729.99",
           :weight "Starting at 3.04 lbs (1.38 kg)",
           :wlan "Intel┬« 9560 802.11AC (2 x 2) & Bluetooth┬« 5.1"}
    diff: - {:orig-price 729.99, :price 372.29}
          + {:orig-price 729.99, :price 372.29}

12/12   100% [==================================================]  ETA: 00:00

Ran 12 tests in 0.100 seconds
68 assertions, 1 failure, 0 errors.
Tests failed.
PS C:\src\logaze-scraper>
```

A numerical value with a decimal point in Clojure defaults to Double. i.e. `{ :price 372.29 }`. I think this can be fixed by either changing to `Double/parseDouble` when reading in the value, or changing the tests to be like `{ :price (float 372.29)}`. I guess by default, Double is the first class citizen in both Java and Clojure, and there's extra work needed to switch to Float.

I would probably also switch `:save-percent` to Double too, now that I'm noticing it.

Reply:
Thanks! I guess I don't really like changing the code for the test but since we're not constrained by memory I guess this is the simple solution :) Thanks for checking this!